app/components/collections/work_row_component.rb

```diff
@@ -0,0 +1,23 @@
+# typed: false
+# frozen_string_literal: true
+
+module Collections
+  # Displays a single table row representing a work in a collection.
+  class WorkRowComponent < ApplicationComponent
+    with_collection_parameter :work
+
+    def initialize(work:)
+      @work = work
+    end
+
+    attr_reader :work
+
+    # Returns the size of the attached files
+    def size
+      number_to_human_size(attached_files.sum(&:byte_size))
```

Comment:
I guess you don't think we need to change the tests to ensure sizes are displayed in the format that `number_to_human_size` returns?

Reply:
not really.  If that were to change, most of the integration tests would fail, right?

Reply:
I don't know. This line is the main change in this PR, right? I guess I'm surprised we needed to add it and it didn't change test behavior.

Reply:
Do we need a test for every last detail?  We could write tests down to the pixel level if we want to, but there are diminishing returns here.  Both before and after we are presenting the same data to the user, this just has a format that is slightly more readable.

Reply:
I'm not advocating for pixel-level testing. What I think we target: test behaviors that our users care about so that we are sure they work as expected and that regressions in those behaviors are prevented (or at least detected). For this PR, despite all the machinery changes, I thought adding `number_to_human_size` was a/the main user-facing change, so I thought it reasonable to ask about whether we're testing it. If it's already tested by feature specs or integration tests, great (though if they were already passing, why/how was this change needed?). If not, then in an instance like this when I don't see a test for the main behavioral change, I can only assume that means we don't care too much if it regresses. True/false?

Reply:
It seems like this could be a point of possible regression like we recently saw in google books with the output of `average` changing, and a test caught that. While it should be a rather simple test, we probably want to at least have a way to catch that kind of change.

Reply:
@aaron-collier What I hear you saying is that we should test that the libraries we are using are not changing behavior.  That is not my understanding of what tests should be for.  The contract is defined by the API and it is up to the library maintainers not to break such contracts.

Reply:
I see this change as akin to #819, which also affects behavior but needs no tests.

Reply:
@jcoyne Thinking about it, I think you're right. In the case of `average` we were tasting the output of our code and the result just happened to change. In this case it would be a test for specifically for `number_to_human_size` and not something we're doing with the result. Good point.

Reply:
The point to my mind was not to test ActiveSupport here; I trust the Rails maintainers to make sure their code is relatively well tested. The point, rather, is that we have a user-facing requirement that file sizes display in a more human-readable form vs. a bare integer, and there's a *lot* of machinery required to satisfy that requirement. The machinery itself is well tested but the way we compose all of this machinery to satisfy this particular requirement may not be.

We do refactorings all the time and it's not unheard of for us to introduce bugs in our refactorings, particularly when requirements are not tested (in e.g., a feature spec). What if in our next refactoring of these components, someone winds up yanking the `number_to_human_size` call? In this case, it would be sufficient to toss a one-line `expect` into an existing feature spec to make sure the size renders in human-readable form. I don't *think* that suggests we need to test every pixel or test every API in every library used in H2.

Reply:
I'm open to being convinced that I am off-base or overthinking this, too.

At any rate, this is a small PR with a change we want to see in sprint 0, so let's commit to resolving this one way or the other today?

Reply:
added a test.